### PR TITLE
Indicate action that failed in YamlColumn

### PR DESCRIPTION
**Pull request by kirs**
Let's indicate whether serialization failed to load or dump of the attribute.

@rafaelfranca @kaspth

### CODE REVIEW

**Comment by maclover7** [comment]
@kirs can you please take a look at the failing Travis build? Thanks!

File: activerecord/lib/active_record/coders/yaml_column.rb [hunk 1]
```diff
@@ -23,16 +23,16 @@ def load(yaml)
...
 
         obj
       end
 
-      def assert_valid_value(obj)
+      def assert_valid_value(obj, action:)
         unless obj.nil? || obj.is_a?(object_class)
           raise SerializationTypeMismatch,
-            "Attribute `#{@attr_name}` was supposed to be a #{object_class}, but was a #{obj.class}. -- #{obj.inspect}"
+            "can't #{action} `#{@attr_name}`: input was supposed to be a #{object_class}, but was a #{obj.class}. -- #{obj.inspect}"
```
Comment (kaspth) [inline activerecord/lib/active_record/coders/yaml_column.rb]: Don't see the need for `: input` :)

File: activerecord/test/cases/coders/yaml_column_test.rb [hunk 2]
```diff
@@ -14,15 +14,15 @@ def test_type_mismatch_on_different_classes_on_dump
...
+        assert_equal %{can't dump `attr_name`: input was supposed to be a Array, but was a String. -- "a"}, error.to_s
       end
 
       def test_type_mismatch_on_different_classes
         coder = YAMLColumn.new("attr_name", Array)
         error = assert_raises(SerializationTypeMismatch) do
           coder.load "--- foo"
         end
-        assert_equal %{Attribute `attr_name` was supposed to be a Array, but was a String. -- "foo"}, error.to_s
+        assert_equal %{can't load `attr_name`: input was supposed to be a Array, but was a String. -- "foo"}, error.to_s
```
Comment (kaspth) [inline activerecord/test/cases/coders/yaml_column_test.rb]: Can we use a different name other than `attr_name` in the tests? My paranoia tells me we could accidentally have missed the string interpolation and would literally print out `attr_name` regardless of the attribute's name :)

**Review by kaspth — COMMENTED** [review]
From the failing build it seems we can't require the `action:` parameter.

There's quite a number of places that use that method or similar:
https://github.com/rails/rails/search?utf8=✓&q=assert_valid_value

**Comment by kirs** [comment]
@kaspth thanks for feedback! I updated the PR.

**Comment by kaspth** [comment]
Thanks!

**Pull request merged**